src/Engines/MeiliSearchEngine.php

```diff
@@ -64,9 +64,9 @@ public function update($models)
             }
 
             return array_merge(
-                [$model->getKeyName() => $model->getScoutKey()],
                 $searchableData,
-                $model->scoutMetadata()
+                $model->scoutMetadata(),
+                [$model->getKeyName() => $model->getScoutKey()],
```

Comment:
@stevebauman - PS. If people are already adding the key in their toSearchableArray() with some value other than what's returned from `getScoutKey()`. Worth considering perhaps? Might also then want to make this change in the MeiliSearchEngine to ensure the `objectID` key is not overridden?

Reply:
this is also something that jumped to my eye. The order here was definitly on purpose the way it was before. is there a specific reason why this was change?

Reply:
@stevebauman can you maybe answer this question?

Reply:
Sorry for the late reply on this one guys! I modified this due to conflicts with the searchable data array. The priority here was intended.

`$searchableData` will contain the model's attributes, including its primary key, alongside other data:

```php
['id' => 1, '...']
```

This means, the custom Scout key will always be overridden (unless the model uses a different primary key). 

Without this, the `array_merge()` structure would be:

```php
array_merge(
    ['id' => 'custom-scout-key:1'],
    ['id' => 1],
    ['metadata'],
)
```

Leading to the custom scout key always being overridden by the primary key.

> @DarronEngelbrechtEdge PS. If people are already adding the key in their toSearchableArray() with some value other than what's returned from getScoutKey(). Worth considering perhaps?

I'm not really sure how we can account for that due to the above scenario...

> Might also then want to make this change in the MeiliSearchEngine to ensure the `objectID` key is not overridden?

Don't you specify the primary key with the MeilisearchEngine? I can't find any reference to objectID inside of Meilisearch's documentation.

Reply:
The implementations the above priority for the `MeiliSearchEngine` and `AlgoliaEngine` differ, as `Algolia` requires a primary key of `objectID`. You cannot change it (to my knowledge).

Since `objectID` is unlikely to be a models primary key, collisions in the `array_merge()` are unlikely, so I haven't adjusted it. Maybe we should to ensure the same priority consistency across the included engines?

**AlgoliaEngine**:

https://github.com/laravel/scout/blob/3352fbc5c071cbd3d5392222363f06cf660ca003/src/Engines/AlgoliaEngine.php#L194


**MeiliSearchEngine**:

https://github.com/laravel/scout/blob/3352fbc5c071cbd3d5392222363f06cf660ca003/src/Engines/MeiliSearchEngine.php#L266

Reply:
> I'm not really sure how we can account for that due to the above scenario... The primary key will likely always be present, so the custom Scout key will never be included in the indexed data. Correct me if I'm wrong?

@stevebauman - You're correct, the array_merge() must decide if it wants to override the `toSearchableArray()` or use the value from there if the user has specified the key returned by `getScoutKeyName()`.

I am in favor of your method change which will override `toSearchableArray()` as this will ensure the rest of the functionality will operate on the correct ID giving the user some protection against bad configurations? Either way it might be worth a mention somewhere in the docs if it's not there already? 

> Don't you specify the primary key with the MeilisearchEngine? I can't find any reference to objectID inside of Meilisearch's documentation.

@stevebauman - Sorry noticed a typo in my previous suggestion, I meant to make the change in the AlgoliaEngine to be sure that the user can't override the `objectID` key in the `toSearchableArray()` but can only modify the value with `getScoutKey()` as intended.

Reply:
> Sorry noticed a typo in my previous suggestion, I meant to make the change in the AlgoliaEngine to be sure that the user can't override the objectID key in the toSearchableArray() but can only modify the value with getScoutKey() as intended.

No worries! Understood, I agree with this change. I've moved the order of this in the `AlgoliaEngine` as well. 👍